### ISSUE TITLE
HTTP request timeout when deleting numerous events from core-data by device name

### ISSUE THREAD

**Issue by MonicaisHer** [edgexfoundry/edgex-go#4036]
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in: 
core-data

### Is this a regression?

 Yes, the previous version in which this bug was not present was: not sure

### Description and Minimal Reproduction [**REQUIRED**]
There are 2,973,356 readings came from a device named "ekuiper" to edgex message bus, and stored in edgex core-data. 
I get a warning from  [edgex-ekuiper](https://github.com/canonical/edgex-ekuiper-snap): 
```
level=warning msg="consumer chan full for request id rule1_EdgeXStream_0" file="edgex/edgex_wrapper.go:119"
```
so I try to clean the consumer edgex core-data by:
```
$ curl -X 'DELETE'   'http://localhost:59880/api/v2/event/device/name/ekuiper' 
It didn't work, and returned the error:
HTTP request timeout
```

## 🔥 Exception or Error
<pre><code>$ curl -X 'DELETE'   'http://localhost:59880/api/v2/event/device/name/ekuiper' 
HTTP request timeout</code></pre>

## 🌍  Your Environment
**Deployment Environment:** Ubuntu 20.04 LTS

**EdgeX Version [**REQUIRED**]:** [latest/edge snap](https://snapcraft.io/edgexfoundry)/[(main branch)](https://github.com/edgexfoundry/edgex-go)

**Anything else relevant?**
There is a workaround that deleting all events by age older than 1 day:
```
$ curl -X 'DELETE'   'http://localhost:59880/api/v2/event/age/86400000000000'
```

**Comment by cloudxxx8** [comment]
thanks, @MonicaisHer , it's really a bug.
the original design is to return 202 Accept response immediately and run the deletion in the background

**Comment by cloudxxx8** [comment]
fixed by #4063
